operator/builtin/parser/csv/csv_test.go

```diff
@@ -61,44 +61,153 @@ func TestCSVParserInvalidType(t *testing.T) {
 
 func TestParserCSV(t *testing.T) {
 	cases := []struct {
-		name         string
-		configure    func(*CSVParserConfig)
-		inputRecord  interface{}
-		outputRecord interface{}
+		name             string
+		configure        func(*CSVParserConfig)
+		inputEntry       entry.Entry
+		outputRecord     interface{}
+		expectBuildErr   bool
+		expectProcessErr bool
```

Comment:
Nice enhancements to the tests.

Consider this optional, but the dynamic processing introduces an additional bit of state that we could test here as well. Based on the description of [csvparser.Reader.FieldsPerRecord](https://pkg.go.dev/encoding/csv#Reader), we can expect potential errors to occur if any record has a different number of fields than the first in a batch. We could enhance the test struct with `inputEntry []entry.Entry`, and add a case for this scenario.

Reply:
Good idea. I switched to slices for input / output, and test all of them using a single operator.